packages/noya-state/src/selectors/gradientSelectors.ts

```diff
@@ -177,3 +178,39 @@ export function isPointerOnGradientLine(state: ApplicationState, point: Point) {
     selectedLayerGradientPoints[selectedLayerGradientPoints.length - 1].point,
   ]);
 }
+
+export function isPointerOnGradientElipseEditor(
+  state: ApplicationState,
+  point: Point,
+) {
+  const selectedLayerGradientPoints = getSelectedGradientStopPoints(state);
+  if (!state.selectedGradient || !selectedLayerGradientPoints) return false;
+  const selectedGradient = getSelectedGradient(
+    getCurrentPage(state),
+    state.selectedGradient,
+  );
+
+  if (!selectedGradient) return false;
+
+  const center = selectedLayerGradientPoints[0].point;
+  const lastPoint =
+    selectedLayerGradientPoints[selectedLayerGradientPoints.length - 1].point;
+
+  const height = distance(center, lastPoint) * 2;
+  const width = height * selectedGradient.elipseLength;
+
+  // Maybe there is a function to do this in a more efficient way
+  const theta =
```

Comment:
We can use an `AffineTransform` here:

```ts
  const position = AffineTransform.rotate(theta, center.x, center.y).applyTo({
    x: center.x - width / 2,
    y: center.y,
  });
```

We should try to share this code with the `SketchFileRenderer`. Maybe:

```ts
function getEllipseEditorPoint(...): Point { ... }
```

Reply:
🤔 I use theta and those points in different ways. The position of the drawn rectangle and the "caption" rectangle are a bit different.